### PR TITLE
refactor: add core.formatLocationFilename, remove op_format_filename

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes https://github.com/denoland/deno/issues/14464

This commit moves "op_format_location" to "core/ops_builtin.rs"
and removes "Deno.core.createPrepareStackTrace" in favor of
"Deno.core.prepareStackTrace".

### CODE REVIEW

**Comment by bartlomieju** [comment]
@nayeemrmn please take a look

**Review by AaronO — COMMENTED** [review]
Moving it into core makes sense, however it looks like this can remain an op (in `ops_builtin.rs`) and doesn't need to be promoted to a v8 binding

File: core/error.rs [hunk 1]
```diff
@@ -431,6 +432,33 @@ pub(crate) fn is_instance_of_error<'s>(
...
+                data_end
+              );
+            }
+          }
+        }
+      }
+    }
+  }
+  file_name.to_string()
+}
```
Comment (AaronO) [inline core/error.rs:460]: Technically your PR doesn't change this code but moves it around, I think we can avoid the deep nested ifs with something like:

```suggestion
pub fn format_file_name(file_name: &str) -> String {
  abbrev_file_name(file_name).unwrap_or_else(|| file_name.to_string())
}

fn abbrev_file_name(file_name: &str) -> Option<String> {
  if file_name.len() <= DATA_URL_ABBREV_THRESHOLD {
    return None;
  }
  let url = Url::parse(file_name).ok()?;
  if url.scheme() != "data" {
    return None;
  }
  let (head, tail) = url.path().split_once(',')?;
  let len = tail.len();
  let start = tail.get(0..20)?;
  let end = tail.get(len - 20..)?;
  Some(format!("{}:{},{}......{}", url.scheme(), head, start, end))
}
```

**Review by AaronO — COMMENTED** [review]
LGTM

**Review by AaronO — APPROVED** [review]

**Pull request merged**